src/test/java/net/masterthought/cucumber/generators/integrations/FeaturesOverviewPageIntegrationTest.java

```diff
@@ -149,6 +150,38 @@ public void generatePage_generatesStatsTableBody() {
         secondRow.getReportLink().hasLabelAndAddress("Second feature", "report-feature_1_1515379431.html");
     }
 
+    @Test
+    public void generatePage_WithSpecifiedQualifier_generatesStatsTableBody() {
+
+        // given
+        String testQualifier = "testQualifier";
+        configuration.setQualifier(FilenameUtils.removeExtension(SAMPLE_JSON), testQualifier);
+        setUpWithJson(SAMPLE_JSON);
+        configuration.addPresentationModes(PresentationMode.PARALLEL_TESTING);
+        page = new FeaturesOverviewPage(reportResult, configuration);
+
+        // when
+        page.generatePage();
+
+        // then
+        DocumentAssertion document = documentFrom(page.getWebPage());
+        TableRowAssertion[] bodyRows = document.getReport().getTableStats().getBodyRows();
+
+        assertThat(bodyRows).hasSize(2);
+
+        TableRowAssertion firstRow = bodyRows[0];
+        firstRow.hasExactValues("1st feature", testQualifier, "10", "0", "0", "0", "0", "10", "1", "0", "1", "1:39.263", "Passed");
+        firstRow.hasExactCSSClasses("tagname", "", "passed", "", "", "", "", "total", "passed", "", "total", "duration", "passed");
```

Comment:
remove from this line to end of section and same for second feature - this allows to focus what is really tested here

Reply:
Done